code/test/tokens.js

```diff
@@ -0,0 +1,501 @@
+/* TODO
```

Comment:
Let's simplify this file (and maybe other files) and remove as much boilerplates / redundant code as possible, and only test for specific features/results related to tokens. We can have a general health check (or state-change check) function that's called in one line (ideally in before/after hooks, rather than repeated in every test), but let's not have the same code block repeating everywhere